frontend/src/components/ui/blockEditor/SpecsInterface.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState } from 'react';
+import { TextInput, Button, Loading, TextArea } from '@carbon/react';
+import { pipelineAtom } from '@/atoms/pipelineAtom';
+import { useImmerAtom } from 'jotai-immer';
+import { Save } from "@carbon/icons-react";
+
+function SpecsInterface({ blockPath }) {
+  const [specs, setSpecs] = useState({});
+  const [flattenedSpecs, setFlattenedSpecs] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [pipeline, setPipeline] = useImmerAtom(pipelineAtom);
+  const serverAddress = "http://localhost:3330";
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch(`${serverAddress}/get-specs`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ blockPath: blockPath })
+    })
+    .then(response => response.json())
+    .then(data => {
+      setSpecs(data);
+      setFlattenedSpecs(flattenSpecs(data)); // Flatten the specs and update state
+      setIsLoading(false);
+    })
+    .catch(error => {
+      console.error('Error fetching specs:', error);
+      setIsLoading(false);
+    });
+  }, [blockPath, serverAddress]);
+
+  const flattenSpecs = (data, prefix = '', path = []) => {
+    let fields = [];
+    Object.keys(data).forEach(key => {
+      if (['connections', 'relays', 'command_line', 'pos_x', 'pos_y', 'pos_z', 'parameters'].includes(key)) {
+        return; // Skip these keys
+      }
+  
+      const newPath = path.concat(key);
+      const newKey = prefix ? `${prefix}.${key}` : key;
+      if (typeof data[key] === 'object' && !Array.isArray(data[key]) && data[key] !== null) {
+        // Recursively call flattenSpecs for nested objects
+        fields = fields.concat(flattenSpecs(data[key], newKey, newPath));
+      } else {
+        // Construct label based on the path
+        let label = key; // Default label
+        if (newPath.includes('inputs') || newPath.includes('outputs')) {
+          const typeLabel = newPath[0].charAt(0).toUpperCase() + newPath[0].slice(1); // Capitalize first letter
+          const varName = newPath[newPath.length - 2]; // Get variable name from the second last item in the path
+          label = `${typeLabel} ${varName} Type`; // Correctly format label
+        }
+        fields.push({ key: newKey, label, value: data[key] });
+      }
+    });
+    return fields;
+  };
+
+  const handleInputChange = (field, value) => {
+    // Update the value in the flattened specs array
+    const newFlattenedSpecs = flattenedSpecs.map(item => {
+      if (item.key === field) {
+        return { ...item, value: value };
+      }
+      return item;
+    });
+    setFlattenedSpecs(newFlattenedSpecs);
+
+    // Update the original specs object
+    const keys = field.split('.');
+    const lastKey = keys.pop();
```

Comment:
possibly you could keep the `newPath` value in the flattened specs so that you don't have to recalculate it here